Build failure: globalarrays

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

$ nix build github:nixos/nixpkgs/nixpkgs-unstable#globalarrays


### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/285842616

### Relevant log output

```console
configure:                 NPROCS=4
configure: 
configurePhase completed in 46 seconds
Running phase: buildPhase
build flags: -j2 SHELL=/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/bash
\
/nix/store/v93bdqf52x2r2lqijv0f7x3yhylwy957-gnused-4.9/bin/sed -e "s/PAPI/WAPI/g" \
       -e "s/pnga_/wnga_/g" \
       -e "/typedef/d" \
           ./global/src/ga-papi.h > global/src/ga-wapi.h
\
/nix/store/v93bdqf52x2r2lqijv0f7x3yhylwy957-gnused-4.9/bin/sed -e "s/PAPI/WAPIDEFS/g" \
           -e "s/^.*p\(nga_.*\)(.*$/#define w\1 p\1/g" \
           -e "/^ /d" \
           -e "/^\/\*/d" \
           -e "/^ \*/d" \
       -e "/typedef/d" \
           ./global/src/ga-papi.h > global/src/ga-wapidefs.h
\
/nix/store/v93bdqf52x2r2lqijv0f7x3yhylwy957-gnused-4.9/bin/sed "/^#/!d" config.h > config.fh
make  all-recursive
make[1]: Entering directory '/build/source'
Making all in comex
make[2]: Entering directory '/build/source/comex'
make  all-am
make[3]: Entering directory '/build/source/comex'
/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/bash ./libtool  --tag=CC   --mode=compile mpicc -DHAVE_CONFIG_H -I.     -I./src-common        -I./src-mpi3     -g -O2 -c -o src-mpi3/comex.lo src-mpi3/comex.c
/nix/store/4fvc5fm8bszmkydng1ivrvr5cbvr1g60-bash-5.2p37/bin/bash ./libtool  --tag=CC   --mode=compile mpicc -DHAVE_CONFIG_H -I.     -I./src-common        -I./src-mpi3     -g -O2 -c -o src-mpi3/groups.lo src-mpi3/groups.c
libtool: compile:  mpicc -DHAVE_CONFIG_H -I. -I./src-common -I./src-mpi3 -g -O2 -c src-mpi3/comex.c -o src-mpi3/comex.o
libtool: compile:  mpicc -DHAVE_CONFIG_H -I. -I./src-common -I./src-mpi3 -g -O2 -c src-mpi3/groups.c -o src-mpi3/groups.o
src-mpi3/comex.c: In function 'comex_free':
src-mpi3/comex.c:3504:20: error: assignment to 'void **' from incompatible pointer type 'long int **' [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wincompatible-pointer-types-Wincompatible-pointer-types8;;]
 3504 |     allgather_ptrs = (long **)malloc(sizeof(void *) * comm_size);
      |                    ^
make[3]: *** [Makefile:1530: src-mpi3/comex.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/build/source/comex'
make[2]: *** [Makefile:1007: all] Error 2
make[2]: Leaving directory '/build/source/comex'
make[1]: *** [Makefile:7417: all-recursive] Error 1
make[1]: Leaving directory '/build/source'
make: *** [Makefile:4461: all] Error 2
```

### Additional context

Full build log: https://cache.nixos.org/log/xpv7hwxfi8qrvwys8506wpbxn9079xrw-globalarrays-5.8.2.drv

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.71, NixOS, 25.05 (Warbler), 25.05.20250116.5df4362`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.11`
 - nixpkgs: `/nix/store/l9nb64iii15y0nr37qrs1cfm6rlpg6gh-source`

### Notify maintainers

@markuskowa 
---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/gl/globalarrays/package.nix
+++ b/pkgs/by-name/gl/globalarrays/package.nix
@@ -30,6 +30,8 @@ stdenv.mkDerivation rec {
     openssh
   ];
 
+  env.NIX_CFLAGS_COMPILE = "-Wno-error=incompatible-pointer-types";
+
   preConfigure = ''
     configureFlagsArray+=( "--enable-i8" \
                            "--with-mpi" \

